### ISSUE TITLE
ImagePullPolicy not working as expected?

### ISSUE THREAD

**Issue by pulpbill** [kubernetes/kubernetes#72401]
I have a cluster with latest version of kops/helm/k8s. I need for the apps to be deployed in case of a gitlab failure, that's why I'm testing setting ImagePullPolicy: IfNotPresent.

I've deployed an app, then modified the manifest to broke the registry url (so it can't pull it) re deployed and I'm getting: 

`Failed to pull image "registry.gitlab.com/our-company/our-app/etc/": rpc error: code = Unknown desc = Error response from daemon: repository registry.gitlab.com/our-company/our-app/etc/ not found: does not exist or no pull access
`
It should be deployed since it's already there. shoulndn't it?

Maybe I'm missing something, I think the docs about this feature could be improved.

Thanks

/triage support

**Comment by liggitt** [comment]
> modified the manifest to broke the registry url

how did you modify it? did you modify the pull spec?

**Comment by pulpbill** [comment]
I have a helm object (a deployment for k8s), I changed at spec -> template -> spec -> containers: -image: and here I just deleted a part of the URL that goes to our registry (gitlab) my intention is for k8s to re deploy with the existing pulled image, hence, imagepullpolicy set to ifnotpresent.

**Comment by liggitt** [comment]
> here I just deleted a part of the URL that goes to our registry

ifnotpresent means "avoid pulling from the registry if an image matching this spec already exists on the node"

by changing the image pull spec, it sounds like you made the pod reference a pull spec that does not exist and had not been pulled to the node yet, so it had to reach out the the registry for the new image

**Comment by pulpbill** [comment]
You're right, sorry, got that part wrong, I'm gonna leave it as it is and just block access to the repo or something like to test if it still can re deploy with same tag.
